src/services/doppler-sites-client.ts

```diff
@@ -0,0 +1,43 @@
+import { AxiosInstance, AxiosStatic } from 'axios';
+
+export interface DopplerSitesClient {
+    getBannerData(lang: string, type: string, page?: string | null): Promise<any>;
+}
+
+export interface Promotions {
+    title: string;
+    functionality: string;
+    description: string;
+    imageUrl: string;
+    backgroundUrl: string;
+}
+
+export class HttpDopplerSitesClient implements DopplerSitesClient {
+    private readonly axios: AxiosInstance;
+    private readonly baseUrl: string;
+
+    constructor({ axiosStatic, baseUrl }: { axiosStatic: AxiosStatic; baseUrl: string }) {
+        this.baseUrl = baseUrl;
+        this.axios = axiosStatic.create({
+          baseURL: baseUrl,
+          withCredentials: true,
+        });
+    }
+
+
+    public async getBannerData(lang: string, type: string, page?: string | null) {
+        const response:any = await this.axios.get(
+            this.baseUrl +`/wp-json/doppler2019/v1/getbanner?filter[lang]=${lang}&filter[type]=${type}&filter[page]=${page || ''}`,
+        );
+        if (!response || !response.data) {
+            throw new Error('Empty Site response');
```

Comment:
❤️

Reply:
What do you think about using error results in place of exceptions as we are doing in the other clients?

In that way, all expected results will be taken into account and documented, it is less error-prone.